### DISCUSSION TITLE
URGENT Support required!!Unable to build docker using SocketViewer/PangolinViewer

### DISCUSSION THREAD

**Discussion by Udaykiran87** [stella-cv/stella_vslam#90] [Q&A] [answered]
Hello OpenVSLAM Community,

I was trying to do docker build for OpenVSLAm using below link:
https://openvslam-community.readthedocs.io/en/latest/docker.html#chapter-docker

I am following **"Instructions for SocketViewer"**

I ran the below command:
**docker build -t openvslam-desktop -f Dockerfile.desktop . --build-arg NUM_THREADS=4**
but i get below error after around 30 minutes of build 

**The command '/bin/sh -c set -x &&   mkdir -p build &&   cd build &&   cmake     -DUSE_SOCKET_PUBLISHER=ON     -DBUILD_EXAMPLES=ON     -DUSE_STACK_TRACE_LOGGER=ON     .. &&   make -j${NUM_THREADS} &&   rm -rf CMakeCache.txt CMakeFiles Makefile cmake_install.cmake example src &&   chmod -R 777 ./*' returned a non-zero code: 1**

I guess it is failing because build folder does not appear to contain CMakeLists.txt.

How do i resolve this error?
Note:
**I get the similar error even for "Instructions for PangolinViewer"**

Hardware and software dependencies are as follows:
**Jetson AGX Xavier
NVIDIA Docker 2.0.3
Ubuntu 18.04**

Any help from the community is highly appreciated.

Thanks and Regards,
Udaykiran Patnaik.

**Comment by ymd-stella** [comment, marked as answer]
Insufficient error log.

**Reply by ymd-stella** [reply]
> Jetson AGX Xavier

This is not included in the test target.
If you want to use it, please make a contribution. I don't own an Jetson AGX Xavier, nor do I have a CI settings for aarch64.

**Reply by Udaykiran87** [reply]
Hello,

The problem was solved.

**Root Cause of the problem.**
As you can see below in the error log, "/openvslam/3rd/FBoW" folder contains no data inside it and hence the build error.

The online document just say that clone using below link.
git clone https://github.com/OpenVSLAM-Community/openvslam.git

But by doing the "/openvslam/3rd/FBoW" contains nothing.

**Solution:**
I had to run below command to get all updated files:
**git submodule update --init**

> Insufficient error log.

CMake Error at src/openvslam/CMakeLists.txt:51 (add_subdirectory):
  The source directory

    /openvslam/3rd/FBoW

  does not contain a CMakeLists.txt file.


-- OpenMP: DISABLED
-- SSE3 for ORB extraction: DISABLED
-- SSE for floating-point operation: DISABLED
-- BoW framework: FBoW (Using submodule)
-- No preference for use of exported glog CMake configuration set, and no hints for include/library directories provided. Defaulting to preferring an installed/exported glog CMake configuration if available.
-- Failed to find installed glog CMake configuration, searching for glog build directories exported with CMake.
-- Failed to find an installed/exported CMake configuration for glog, will perform search for installed glog components.
-- Found Glog: /usr/include  
-- Stack trace logger: ENABLED
-- Google Perftools: DISABLED
-- Viewer for examples: PangolinViewer
-- Configuring incomplete, errors occurred!
See also "/openvslam/build/CMakeFiles/CMakeOutput.log".
See also "/openvslam/build/CMakeFiles/CMakeError.log".
The command '/bin/sh -c set -x &&   mkdir -p build &&   cd build &&   cmake     -DUSE_PANGOLIN_VIEWER=ON     -DBUILD_EXAMPLES=ON     -DUSE_STACK_TRACE_LOGGER=ON     .. &&   make -j${NUM_THREADS} &&   rm -rf CMakeCache.txt CMakeFiles Makefile cmake_install.cmake example src &&   chmod -R 777 ./*' returned a non-zero code: 1
